CMakePresets.json

```diff
@@ -48,13 +48,16 @@
         "strategy": "external"
       },
       "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
```

Comment:
Visual Studio can switch the build type in the IDE or with `cmake --build . --config Debug`. It's not correct to add it here.

Reply:
I know but on my Windows 11 with 7800X3D the `Release` mode has no effect at all. The application is always compiled in `Debug` mode. Same result compiling with both VS and the standalone cmake (external cmake tool)

Reply:
`CMAKE_BUILD_TYPE` does not work for multi-config generators.

From the official CMake documentation.
> For multi-configuration generators like [Visual Studio](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#visual-studio-generators), [Xcode](https://cmake.org/cmake/help/latest/generator/Xcode.html#generator:Xcode), and [Ninja Multi-Config](https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html#generator:Ninja%20Multi-Config), the configuration is chosen by the user at build time and [CMAKE_BUILD_TYPE](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html#variable:CMAKE_BUILD_TYPE) is ignored.

I don't know what you mean when say it had no effect. I just tested it and it worked as expected.

Reply:
I mean that from the IDE I cannot select the compile config `Release` or `Debug` (even removing `"CMAKE_BUILD_TYPE": "Debug"` from `CMakePresets.json`). If I select `rpcs3.exe` from the Solution Explorer panel there is no possibility to select  `Release` or `Debug`. See attached picture.

![vs_cmake](https://github.com/user-attachments/assets/1d1b24e2-8721-45fb-82e2-3bdf8c57ab49)

If I provide `"CMAKE_BUILD_TYPE": "Debug"` or `"CMAKE_BUILD_TYPE": Release"` (as it is done for `gcc` and `clang`) the application is always compiled in `Debug`. From command line, I will verify  the suggested `--config Debug` option

Reply:
the `--config Debug`  and `--config Release` options seems to work from command line. I tested the standalone `cmake` tool (not the MSVC's cmake).
But how to set it from the IDE?

Reply:
To set it from the IDE you need to define a build preset for each configuration. Something like this:
```
    "buildPresets": [
        {
            "name": "msvc-debug",
            "configurePreset": "msvc"
            "configuration": "Debug"
        },
        {
            "name": "msvc-release",
            "configurePreset": "msvc",
            "configuration": "Release"
        },
    ]
```

Reply:
Verify that your sln and/or vcxproj file is not broken. All configured targets should appear in the dropdowns at the top for your CPU arch and build type.

Reply:
He is not opening the "sln", he is opening the "rpcs3" directory.

Reply:
> To set it from the IDE you need to define a build preset for each configuration. Something like this:
> 
> ```
>     "buildPresets": [
>         {
>             "name": "msvc-debug",
>             "configurePreset": "msvc"
>             "configuration": "Debug"
>         },
>         {
>             "name": "msvc-release",
>             "configurePreset": "msvc",
>             "configuration": "Release"
>         },
>     ]
> ```

I saw that `buildPresets` were missing in the `CMakePresets.json`. I created instead two configPresets for `msvc-debug` and `msvc-release` but it didn't work (those have been added in the **Configuration** drop-down menu with `gcc` and `clang` and as I said both have been compiled in Debug mode for the reason you reported). I will add the `buildPresets`, they should not conflict with `gcc` and `clang` configs.

Reply:
fixed